server/lib/chat/chat-service.test.ts

```diff
@@ -0,0 +1,1505 @@
+import { RequestSessionLookup } from '../websockets/session-lookup'
+import { UserSocketsManager } from '../websockets/socket-groups'
+import {
+  clearTestLogs,
+  createFakeNydusServer,
+  InspectableNydusClient,
+  NydusConnector,
+} from '../websockets/testing/websockets'
+import { TypedPublisher } from '../websockets/typed-publisher'
+import {
+  addMessageToChannel,
+  addUserToChannel,
+  banAllIdentifiersFromChannel,
+  banUserFromChannel,
+  ChatMessage,
+  countBannedIdentifiersForChannel,
+  createChannel,
+  deleteChannelMessage,
+  findChannelByName,
+  getChannelInfo,
+  getMessagesForChannel,
+  getUserChannelEntryForUser,
+  getUsersForChannel,
+  isUserBannedFromChannel,
+  removeUserFromChannel,
+  TextMessageData,
+  updateUserPermissions,
+  UserChannelEntry,
+} from './chat-models'
+import ChatService, { getChannelPath, getChannelUserPath } from './chat-service'
+
+const flagsMock = flags as { CAN_LEAVE_SHIELDBATTERY_CHANNEL: boolean }
+
+jest.mock('../../../common/flags', () => ({
+  __esModule: true,
+  CAN_LEAVE_SHIELDBATTERY_CHANNEL: false,
+}))
+
+// NOTE(2Pac): We don't bother mocking this since all the DB methods are mocked already and we're
+// not doing anything special with this in chat service except pass it as an argument to the model
+// methods.
+const dbClient = {} as DbClient
+
+jest.mock('../db/transaction', () =>
+  jest.fn(async next => {
+    await next(dbClient)
+  }),
+)
+
+jest.mock('../models/permissions')
+jest.mock('../users/user-identifiers')
+jest.mock('../users/user-model')
+jest.mock('./chat-models', () => ({
+  getChannelsForUser: jest.fn().mockResolvedValue([]),
+  getUsersForChannel: jest.fn().mockResolvedValue([]),
+  getUserChannelEntryForUser: jest.fn(),
+  createChannel: jest.fn(),
+  addUserToChannel: jest.fn(),
+  addMessageToChannel: jest.fn(),
+  getMessagesForChannel: jest.fn().mockResolvedValue([]),
+  deleteChannelMessage: jest.fn(),
+  removeUserFromChannel: jest.fn(),
+  updateUserPermissions: jest.fn(),
+  countBannedIdentifiersForChannel: jest.fn(),
+  banUserFromChannel: jest.fn(),
+  banAllIdentifiersFromChannel: jest.fn(),
+  isUserBannedFromChannel: jest.fn(),
+  getChannelInfo: jest.fn().mockResolvedValue([]),
+  findChannelByName: jest.fn(),
+  searchChannelsAsAdmin: jest.fn().mockResolvedValue([]),
+}))
+
+describe('chat/chat-service', () => {
+  let nydus: NydusServer
+  let chatService: ChatService
+  let connector: NydusConnector
+
+  const shieldBatteryChannel: ChannelInfo = {
+    id: makeSbChannelId(1),
+    name: 'ShieldBattery',
+    private: false,
+    official: true,
+  }
+  const testChannel: ChannelInfo = {
+    id: makeSbChannelId(2),
+    name: 'test',
+    private: false,
+    official: false,
+  }
+
+  const user1: SbUser = { id: makeSbUserId(1), name: 'USER_NAME_1' }
+  const user2: SbUser = { id: makeSbUserId(2), name: 'USER_NAME_2' }
+
+  const userPermissions: SbPermissions = {
```

Comment:
Move this value to the common permissions file and call it `DEFAULT_PERMISSIONS`. We already have like 10 files that have to be updated whenever a new permission is added, I don't want another

(That value should be readonly in TS, so this would become `const userPermissions = { ...DEFAULT_PERMISSIONS }` if you need this as an actual value somewhere)